src/model.rs

```diff
@@ -29,12 +36,8 @@ impl Model {
             .into_iter()
             .map(|(ngram, _count)| ngram);
 
-        let mut ngram_ranks = HashMap::new();
-        for (rank, ngram) in ngrams.enumerate() {
-            ngram_ranks.insert(ngram, rank);
-        }
-
-        Model { ngram_ranks: ngram_ranks }
+        // Nicer way to build a hash map.
+        Model { ngram_ranks: ngrams.enumerate().map(|(a, b)| (b, a)).collect() }
```

Comment:
Note: It may be more efficient to store these in a `Vec<String>` and binary search.